app/controllers/application_controller/filter.rb

```diff
@@ -307,11 +307,11 @@ def adv_search_button
             {:model => ui_lookup(:model => @edit[@expkey][:exp_model]),
              :name => @edit[:new_search_name]})
           @edit[@expkey].select_filter(s)
-          @edit[:new_search_name] = @edit[:adv_search_name] = @edit[@expkey][:exp_last_loaded][:description]
+          @edit[:new_search_name] = @edit[@expkey][:exp_model] != "Storage" ? @edit[:adv_search_name] = @edit[@expkey][:exp_last_loaded][:description] : @edit[:new_search_name]
```

Comment:
I believe this can be solved by some other (and hopefully better) solutions. If you look at the original line and test it, you will realize that the result will be that `@edit[:new_search_name]` will be `nil` which is not good. I decided to do it like that to leave it as it was for other cases (I mean not "Storage") because hopefully it was a reason for how it was like that for so long. Maybe somebody could test it if it could be changed generally. I didnt want to break functionality in other places with this. If you have a better working solution, tell me.